### DISCUSSION TITLE
"no main manifest attribute" Kotlin, Gradle

### DISCUSSION THREAD

**Discussion by zax71** [Minestom/Minestom#1831] [Help] [answered]
Not sure why this error is occurring, I thought it was because the main class wasn't defined in Gradle, but it seems to be defined correctly. My `main.kt` file and `build.gradle.kts` are available [here](https://paste.gg/p/anonymous/47eac889e39443dbb3be2b9dcf54a8cc).

File Structure:

![image](https://user-images.githubusercontent.com/67716263/232334464-9f8a175a-a7a0-4c0b-9e23-837d59bbe8bf.png)

**Comment by tomhula** [comment]
You need to set main class in the application block. https://docs.gradle.org/current/userguide/application_plugin.html

**Reply by zax71** [reply]
That didn't seam to fix it

**Comment by emortaldev** [comment, marked as answer]
https://wiki.minestom.net/setup/your-first-server#build-the-server-jar
which on .kts is equivalent to
```kotlin
tasks {
    named<ShadowJar>("shadowJar") {
        manifest {
            attributes (
                "Main-Class" to "me.zax71.stomKor.MainKt",
                "Multi-Release" to true
            )
        }
    }
}
```
(Since you're using Kotlin, you have to append "Kt")
btw your package name should be lowercase

**Reply by zax71** [reply]
I switched to Java for that project now, but i'll mark this as the solution as it seams logical

**Reply by ThatOneToast** [reply]
I am facing the same issue, and this did not solve it for me.

**Reply by ThatOneToast** [reply]
Adding the application plugin, and then in your build gradle add

```
plugins {
    kotlin("jvm") version "2.0.0"
    id("com.github.johnrengelman.shadow") version "8.1.1"
    id("application")
}

application {
    mainClass = "org.example.MainKt"
}

tasks {
    named<ShadowJar>("shadowJar") {
        manifest {
            attributes(
                "Main-Class" to "org.example.MainKt",
                "Multi-Release" to true
            )
        }
    }
}
```

This solved it for me.
